odata-protocol/11.4 Data Modification.md

```diff
@@ -701,7 +701,7 @@ through a navigation property path binding [#OData-CSDL#NavigationPropertyPathBi
 or through the presence of a context URL within the payload;
 if the service is unable to determine it, it MUST fail the request.
 
-Upserts are not supported against entities whose keys' values are
+Upsert requests MUST NOT be treated as inserts of entities whose keys' values are
```

Comment:
it's really can not -- there is no way to specify the url of an entity whose key is generated by the service and does not yet exist.  The MUST comes in the following line.